client/src/components/Header.tsx

```diff
@@ -0,0 +1,115 @@
+import AppBar from "@mui/material/AppBar";
+import Link from "@mui/material/Link";
+import InputBase from "@mui/material/InputBase";
+import SearchIcon from "@mui/icons-material/Search";
+import { styled, alpha } from "@mui/material/styles";
+import Toolbar from "@mui/material/Toolbar";
+import MenuItem from "@mui/material/MenuItem/MenuItem";
+import Menu from "@mui/material/Menu";
+import React from "react";
+import AccountCircle from "@mui/icons-material/AccountCircle";
+import IconButton from "@mui/material/IconButton/IconButton";
+
+const Search = styled("div")(({ theme }) => ({
+  position: "relative",
+  borderRadius: theme.shape.borderRadius,
+  backgroundColor: alpha(theme.palette.common.white, 0.15),
+  "&:hover": {
+    backgroundColor: alpha(theme.palette.common.white, 0.25),
+  },
+  marginLeft: "auto",
+  marginRight: "auto",
+  width: "auto%",
+}));
+
+const SearchIconWrapper = styled("div")(({ theme }) => ({
+  padding: theme.spacing(0, 2),
+  height: "100%",
+  position: "absolute",
+  pointerEvents: "none",
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+  color: "inherit",
+  "& .MuiInputBase-input": {
+    padding: theme.spacing(1, 1, 1, 0),
+    paddingLeft: `calc(1em + ${theme.spacing(8)})`,
+    width: "50ch",
+
+    /* Resize searchbar based on screen size */
+    [theme.breakpoints.up("lg")]: {
+      width: "90ch",
+    },
+    [theme.breakpoints.between("sm", "md")]: {
+      width: "25ch",
+    },
+    [theme.breakpoints.down("sm")]: {
+      width: "9ch",
+    },
+  },
+}));
+
+function AccountMenu() {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
```

Comment:
You don't want to store elements in state. You can reference an element by id, or conditionally render it.